### PR TITLE
Use for-each instead of for-loop and while-loop

### PR DESCRIPTION
**Pull request by wreulicke**
Use for-each instead of for-loop and while-loop

----

Make sure these boxes are checked before submitting your PR -- thank you!

- [x] Added an entry into `CHANGELOG.md` if you have changed SpotBugs code

### CODE REVIEW

File: spotbugs/src/main/java/edu/umd/cs/findbugs/NewResults.java [hunk 1]
```diff
@@ -42,14 +41,12 @@ public NewResults(SortedBugCollection origCollection, SortedBugCollection newCol
...
-
-        for (Iterator<BugInstance> i = newCollection.iterator(); i.hasNext();) {
-            BugInstance bugInstance = i.next();
-
-            if (!origCollection.contains(bugInstance)) {
-                result.add(bugInstance);
-            }
-        }
+	
+		for (BugInstance bugInstance : newCollection) {
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/NewResults.java]: Inconsistent indentation (tabs vs spaces)

File: spotbugs/src/main/java/edu/umd/cs/findbugs/SortedBugCollection.java [hunk 2]
```diff
@@ -617,11 +617,10 @@ public void writeEpilogue(XMLOutput xmlOutput) throws IOException {
     private void writeBugPatterns(XMLOutput xmlOutput) throws IOException {
         // Find bug types reported
         Set<String> bugTypeSet = new HashSet<>();
-        for (Iterator<BugInstance> i = iterator(); i.hasNext();) {
-            BugInstance bugInstance = i.next();
-            BugPattern bugPattern = bugInstance.getBugPattern();
-            bugTypeSet.add(bugPattern.getType());
-        }
+		for (BugInstance bugInstance : this) {
```
Comment (ThrawnCA) [inline spotbugs/src/main/java/edu/umd/cs/findbugs/SortedBugCollection.java]: Inconsistent indentation (tabs vs spaces)

**Comment by wreulicke** [comment]
> please indent with spaces to match the surrounding code.

Sorry, fixed.

**Comment by wreulicke** [comment]
Fixed conflicts.

Could you review again, please? @ThrawnCA

**Comment by wreulicke** [comment]
@ThrawnCA 

Rebased.
Could you review again?

**Comment by wreulicke** [comment]
Sorry, I will fix CI

**Comment by wreulicke** [comment]
> continuous-integration/travis-ci/push

ah, This CI is failed because I pushed this repository (not is fork.)
`continuous-integration/travis-ci/push` is ignorable.

**Comment by wreulicke** [comment]
@ThrawnCA 
Could you review again?

**Comment by wreulicke** [comment]
@ThrawnCA Thanks! 🍺

**Pull request merged**